testfixture/testfixture_test.go

```diff
@@ -0,0 +1,38 @@
+package testfixture
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestRunScriptWithInputFile(t *testing.T) {
+	// Create a temporary script file.
+	scriptContent := `#!/bin/bash
+					  echo "Hello, World!"`
+
+	scriptFile, err := os.CreateTemp("", "testscript-*.sh")
+	if err != nil {
+		t.Fatalf("Failed to create temp file: %v", err)
+	}
+	defer os.Remove(scriptFile.Name())
+
+	_, err = scriptFile.Write([]byte(scriptContent))
+	if err != nil {
+		t.Fatalf("Failed to write to temp file: %v", err)
+	}
+	scriptFile.Close()
```

Comment:
If this failure is reached, isn't there a possibility of leaving the `scriptFile` handle open?

If so we should probably call `defer scriptFile.Close()` right after file creation. You should also be able to use `Sync()` instead of the current call to `Close()`.

Reply:
Here I am not checking the persistence of content to a file, But testing the actual function that takes stdin for the script it executes, not going to use these file anywhere else. so `close()` is sufficient for this test.